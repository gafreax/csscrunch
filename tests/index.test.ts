import { describe, it, expect } from 'vitest'
import { readFile } from 'node:fs/promises'
import path from 'path'
import fs from 'fs'

import compile from '../src/index'

const cssList = fs.readdirSync(path.join(__dirname, '__css__')).filter(file => file.endsWith('.css'))
describe('compile', () => {
  it.each(cssList)('compiles CSS %s', async (cssFile: string) => {
    const css = await readFile(path.join(__dirname, '__css__', cssFile), 'utf-8')
    const result = compile(css)
    expect(result).toMatchSnapshot()
  })
})
