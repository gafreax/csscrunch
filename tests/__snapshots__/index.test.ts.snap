// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`compile > compiles CSS 1`] = `
".a{padding:ed lor:red or:red olor:red r:red lor:red or:red olor:red :red lor:red or:red olor:red or:red lor:red or:red olor:red ed lor:red or:red olor:red r:red lor:red or:red olor:red :red lor:red or:red olor:red or:red lor:red or:red olor:red d lor:red or:red olor:red r:red lor:red or:red olor:red :red lor:red or:red olor:red or:red lor:red or:red olor:red red lor:red or:red olor:red r:red lor:red or:red olor:red :red lor:red or:red olor:red or:red lor:red or:red olor:red lor:red or:red olor:red r:red lor:red or:red olor:red :red lor:red or:red olor:red or:red lor:red or:red olor:red ed lor:red or:red olor:red r:red lor:red or:red olor:red :red lor:red or:red olor:red or:red lor:red or:red olor:red d lor:red or:red olor:red r:red lor:red or:red olor:red :red lor:red or:red olor:red or:red lor:red or:red olor:red red lor:red or:red olor:red r:red lor:red or:red olor:red :red lor:red or:red olor:red or:red lor:red or:red olor:red lor:red or:red olor:red r:red lor:red or:red olor:red :red lor:red or:red olor:red or:red lor:red or:red olor:red ed lor:red or:red olor:red r:red lor:red or:red olor:red :red lor:red or:red olor:red or:red lor:red or:red olor:red d lor:red or:red olor:red r:red lor:red or:red olor:red :red lor:red or:red olor:red or:red lor:red or:red olor:red red lor:red or:red olor:red r:red lor:red or:red olor:red :red lor:red or:red olor:red or:red lor:red or:red olor:red d lor:red or:red olor:red r:red lor:red or:red olor:red :red lor:red or:red olor:red or:red lor:red or:red olor:red ed lor:red or:red olor:red r:red lor:red or:red olor:red :red lor:red or:red olor:red or:red lor:red or:red olor:red d lor:red or:red olor:red r:red lor:red or:red olor:red :red lor:red or:red olor:red or:red lor:red or:red olor:red red lor:red or:red olor:red r:red lor:red or:red olor:red :red lor:red or:red olor:red or:red lor:red or:red olor:red;}.blu-class{padding:00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 r:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 :#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 or:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 r:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 :#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 or:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 0ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 r:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 :#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 or:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 #00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 r:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 :#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 or:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 r:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 :#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 or:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 r:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 :#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 or:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 0ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 r:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 :#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 or:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 #00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 r:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 :#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 or:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 f00 lor:#00ff00 or:#00ff00 olor:#00ff00 r:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 :#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 or:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 r:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 :#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 or:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 0ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 r:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 :#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 or:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 #00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 r:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 :#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 or:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 0ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 r:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 :#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 or:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 r:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 :#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 or:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 0ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 r:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 :#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 or:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 #00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 r:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 :#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 or:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00;}.test-include{padding:0px in:0px n:0px gin:0px 0px in:0px n:0px gin:0px px in:0px n:0px gin:0px :0px in:0px n:0px gin:0px x in:0px n:0px gin:0px 0px in:0px n:0px gin:0px px in:0px n:0px gin:0px :0px in:0px n:0px gin:0px in:0px n:0px gin:0px 0px in:0px n:0px gin:0px px in:0px n:0px gin:0px :0px in:0px n:0px gin:0px px in:0px n:0px gin:0px 0px in:0px n:0px gin:0px px in:0px n:0px gin:0px :0px in:0px n:0px gin:0px;}.c{padding:lor:red lor:red or:red olor:red r:red lor:red or:red olor:red :red lor:red or:red olor:red or:red lor:red or:red olor:red;}.e{padding:lor:blueviolet lor:blueviolet or:blueviolet olor:blueviolet r:blueviolet lor:blueviolet or:blueviolet olor:blueviolet :blueviolet lor:blueviolet or:blueviolet olor:blueviolet or:blueviolet lor:blueviolet or:blueviolet olor:blueviolet;}.b,#example, .esempio{padding:lor:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 r:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 :#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00 or:#00ff00 lor:#00ff00 or:#00ff00 olor:#00ff00;}
"
`;
